cirq/ops/linear_combinations.py

```diff
@@ -315,6 +316,90 @@ def copy(self) -> 'PauliSum':
         factory = type(self)
         return factory(self._linear_dict.copy())
 
+    def expectation_from_wavefunction(
+            self,
+            state: np.ndarray,
+            qubit_map: Dict[raw_types.Qid, int]) -> float:
+        """Evaluate the expectation of this PauliSum given a wavefunction.
+
+        See `PauliString.expectation_from_wavefunction`.
+        """
+        if any([abs(p.coefficient.imag) > 0.0001 for p in self]):
+            raise NotImplementedError(
+                "Cannot compute expectation value of a non-Hermitian "
+                "PauliString <{}>. Coefficient must be real.".format(self))
+
+        if not isinstance(qubit_map, dict) or not all(
```

Comment:
Don't require that it be a `dict` (e.g. a `defaultdict` would be fine). I think `isinstance(x, Mapping)` from `typing` works more generally.